rudder-lang/src/parser.rs

```diff
@@ -260,10 +283,10 @@ fn penum(i: PInput) -> PResult<PEnum> {
         {
             metadata: pmetadata_list; // metadata unsupported here, check done after 'enum' tag
             global: opt(tag("global")); // TODO at least one space here
-            e:      tag("enum"); // TODO at least one space here
-            _fail: or_fail(verify(peek(anychar), |_| metadata.is_empty()), || PErrorKind::UnsupportedMetadata(metadata[0].key.into()));
-            name:   or_fail(pidentifier, || PErrorKind::InvalidName(e));
-            b:      tag("{"); // do not fail here, it could still be a mapping
+            e:      or_err(tag("enum"), || PErrorKind::ExpectedReservedWord("enum")); // TODO at least one space here
+            _fail:  or_err(verify(peek(anychar), |_| metadata.is_empty()), || PErrorKind::UnsupportedMetadata(metadata[0].key.into()));
```

Comment:
why did you remove or_fail ? 
it seems to me that if the user has already written "enum blah" an invalid blah is a fail.
and if it is not, please rename the variable too

Reply:
Just checked, this was a mistake: I needed to have every enum error non-terminated for testing purpose and forgot to revert it. Fixed